### ISSUE TITLE
Using pre-trained weights in YOLOv3

### ISSUE THREAD

**Issue by gnoya** [AlexeyAB/darknet#1694]
Hi, I want to train yolov3 to detect custom objects, but my dataset for **detection** is small (900 images, around 1300x800 px). But, I have a dataset for **classification** with about 50k images (cropped objects only, around 120x120 px). Both have the same 43 classes.

I read that if I pre train a CNN to classify and then use those pretrained weights to train for detection I would get better results. So, I had an idea, take the Darknet53 weights and cfg from here https://pjreddie.com/darknet/imagenet/, train for **classification** with the 50k image dataset, and then use those weights as my new pre-trained weights for yolov3 to train for **detection**, but stopping the backpropagation at shallow layers so my training doesn't take that long (also, shallower layer contain feature detectors that I probably don't need to modify that much).

So I have some doubts about this process. It is okay to download those previously mentioned weights and use them directly to train for classification? If so, how do I adapt them to train in yolov3 for detection? Because I have seen "darknet53.conv.74", and I don't get the difference between that .conv.74 and .weight files. I saw some post about the **"partial"** command, but I didn't understand the usage.

**Comment by AlexeyAB** [comment]
@gnoya Hi,

Yes, you can train https://github.com/AlexeyAB/darknet/blob/master/cfg/darknet53.cfg as described here https://pjreddie.com/darknet/imagenet/ on your classification dataset to get new `darknet53.weights` file.

Then do 
`./darknet partial cfg/darknet53.cfg darknet53.weights darknet53.conv.74 74`
to get new pre-trained weights file `darknet53.conv.74` with only first 74-layers from your trained `darknet53.weights` file.

Then train your model with this new file `darknet53.conv.74`
`./darknet detector train data/obj.data cfg/yolov3-obj.cfg darknet53.conv.74`

----

If you want to freeze pre-trained weights, then add `stopbackward=1` here: https://github.com/AlexeyAB/darknet/blob/6c8642b3eb19b68a11f3fae6a55bbfd528a800d3/cfg/yolov3.cfg#L544

**Comment by gnoya** [comment]
Thank you very much! I have another question, there are no instructions about training in https://pjreddie.com/darknet/imagenet/, I image I could follow this https://pjreddie.com/darknet/train-cifar/ but using my dataset, is that right? Do I have to follow that label formatting or is it different?

**Comment by AlexeyAB** [comment]
You can use the same format of labels as for Imagenet/Cifar.

**Comment by gnoya** [comment]
Thanks again! Last question, I need some information about the data augmentation you are using, for example, how many different images are generated, or info like that.

**Comment by AlexeyAB** [comment]
Will be generated number of images `batch * iterations`. I.e. each time each image will be randomly changed.

You can see: https://github.com/AlexeyAB/darknet/blob/6c8642b3eb19b68a11f3fae6a55bbfd528a800d3/src/data.c#L110-L136

https://github.com/AlexeyAB/darknet/blob/6c8642b3eb19b68a11f3fae6a55bbfd528a800d3/src/image.c#L1454-L1473

**Comment by gnoya** [comment]
Thanks! You can close the issue.
